### DISCUSSION TITLE
Themes without qt-materials or how to use/install it under bookworm without breaking external packages

### DISCUSSION THREAD

**Discussion by OderDochNicht** [evilsocket/opensnitch#999] [Q&A] [answered]
This is for a bare metal installation of debian bookworm, done with a minimal system via netinst.iso. It shall manage some VMs via libvirt and control outgoing traffic. For security reasons it shall stay as minimalistic as possible. Therefore initially there is no python3-full and pip3 or qt-materials. 
After installation of opensnitch the ui is quite unusable. Buttons are not visible and can be found only with clicking to the area where they might be.
After installation of pip3 I tried 
`pip3 install qt-material `
and got the warning:
```
error: externally-managed-environment
× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.
    
    See /usr/share/doc/python3.11/README.venv for more information.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
```
Playing around with themes in xfce4 doesn't help either.
Is it really necessary to install a monster like qt-materials to get opensnitch-ui working?
And how to do that without breaking system packages?

**Comment by gustavo-iniguez-goya** [comment]
Hi @OderDochNicht ,

> After installation of opensnitch the ui is quite unusable. Buttons are not visible and can be found only with clicking to the area where they might be.

This is not a problem of OpenSnitch, but your installation/configuration.

See if these instructions help: https://github.com/evilsocket/opensnitch/wiki/GUI-known-problems#no-icons-on-the-gui

**Reply by OderDochNicht** [reply]
Hi @gustavo-iniguez-goya,

thanks for your fast response.

Here the results:


1.    XDG_CURRENT_DESKTOP=GNOME opensnitch-ui
`Themes not available. Install qt-material if you want to change GUI's appearance: pip3 install qt-material. 
~ OpenSnitch GUI - 1.6.0 ~ 
protobuf: 4.21.12 - grpc: 1.51.1 
-------------------------------------------------- ` 

2.    qt5ct installed and QT_QPA_PLATFORMTHEME=qt5ct opensnitch-ui
`Themes not available. Install qt-material if you want to change GUI's appearance: pip3 install qt-material. 
~ OpenSnitch GUI - 1.6.0 ~ 
protobuf: 4.21.12 - grpc: 1.51.1 
-------------------------------------------------- `

So, no luck. Is there no way without qt-material?

I don't use GNOME but xfce4.

**Reply by gustavo-iniguez-goya** [reply]
Themes are optional. That message only indicates that if you want to use opensnitch's theming functionality you need to install qt-material.
Anyway, qt-material themes only changes widgets' appearance, but no the icons.

If you don't see the icons on the GUI you'll need some additional packages. maybe `qt5-gtk-platformtheme` ?

Taking a look at the Qt's code, it expects some enviromental variables: XDG_SESSION_TYPE, XDG_CURRENT_DESKTOP, XDG_SESSION_DESKTOP
https://codebrowser.dev/qt5/qtbase/src/gui/kernel/qguiapplication.cpp.html#1406

For example, on i3 Window Manager these variables are not set by default, therefore the icons don't appear on the GUI. But if you launch the GUI with `~ $ XDG_CURRENT_DESKTOP=gnome opensnitch-ui` then they appear.

On i3 I had to follow the steps described here to configure the icon theme with `qt5ct`: https://github.com/evilsocket/opensnitch/wiki/GUI-known-problems#no-icons-on-the-gui

**Reply by OderDochNicht** [reply, marked as answer]
Something went wrong when I did
`
👉 Alternatively, install qt5ct, launch it an go to the "Icons Theme" tab. Select an icon theme an click on Apply.
Then launch the GUI as follow: ~ $ QT_QPA_PLATFORMTHEME=qt5ct opensnitch-ui`

After a second try (maybe after a logout and login) it was fixed and the icons showed up in opensnitch-ui.

Thanks a lot for all your help!

**Reply by trsmcse** [reply]
Hi OrderDochNicht! I followed and tried according your discussion, but unfortunately I'm unable to solve with current situation. I have shared screenshot that I faced the issue. I'm using ParrotOS - KDE Plasma Desktop Environment. 

![OpenSnitch-UI](https://github.com/evilsocket/opensnitch/assets/137600554/9b211ac7-92fa-4f0a-8b4d-95402f6009b4)

**Reply by trsmcse** [reply]
![qt-material](https://github.com/evilsocket/opensnitch/assets/137600554/c15f45cc-d40d-493a-b5f4-2cb14909743d)

**Reply by gustavo-iniguez-goya** [reply]
hey @trsmcse , did you try launching the GUI as your regular user? The GUI doesn't need to run as root.

On the other hand, verify that qt-material is installed for the user `parrotos`: `~ $ pip3 list installed | grep qt-material`

**Reply by trsmcse** [reply]
It's working. Thanks a lot for your help.
